framework/contracts/native/ibc-client/src/commands.rs

```diff
@@ -307,19 +307,74 @@ pub fn execute_send_query(
     env: Env,
     info: MessageInfo,
     host_chain: String,
-    query: QueryRequest<Empty>,
+    query: QueryRequest<ModuleQuery>,
     callback_info: CallbackInfo,
 ) -> IbcClientResult {
     let host_chain = ChainName::from_str(&host_chain)?;
 
+    let ibc_client_callback = IbcClientCallback::ModuleRemoteQuery {
+        callback_info,
+        sender_address: info.sender.to_string(),
+        query: query.clone(),
+    };
     let callback_request = CallbackRequest {
         receiver: env.contract.address.to_string(),
-        msg: to_json_binary(&IbcClientCallback::ModuleRemoteQuery {
-            callback_info,
-            sender_address: info.sender.to_string(),
-            query: query.clone(),
-        })
-        .unwrap(),
+        msg: to_json_binary(&ibc_client_callback).unwrap(),
+    };
+
+    // Convert it to polytone(empty) query or send module to module query action if custom
+    let query: QueryRequest<Empty> = match query {
+        QueryRequest::Custom(ModuleQuery { target_module, msg }) => {
+            let cfg = CONFIG.load(deps.storage)?;
+            let ibc_infra = IBC_INFRA.load(deps.storage, &host_chain)?;
+            let remote_ibc_host = ibc_infra.remote_abstract_host;
+            // Query the sender module information
+            let module_info = cfg
+                .version_control
+                .module_registry(deps.as_ref())?
+                .module_info(info.sender.clone())?;
+
+            // We need additional information depending on the module type
+            let source_module = match module_info.reference {
```

Comment:
This is not necessary here. Let's use target_module as an InstalledModuleIdentification type and let the user query any app associated with any account on the remote chain.

Reply:
Thanks!